### ISSUE TITLE
Workers statistics empty (all zeroes)

### ISSUE THREAD

**Issue by xme** [OpenCTI-Platform/opencti#1401]
## Description

Upgraded to the latest version (through dockers) and now my worker's statistics page report all zeroes
(but data are properly ingested)

## Environment

1. OS: Ubuntu 20.04
2. OpenCTI version: 4.5.4
3. OpenCTI client: frontent
4. Other environment details: -

## Reproducible Steps

Steps to create the smallest reproducible scenario:

Goto page "Data" -> "Connectors"

## Expected Output

Workers statistics

## Actual Output

All zeroes 

## Additional information

-

**Comment by richard-julien** [comment]
Hi @xme ,

Workers statistics are directly get from rabbitMQ Management API.
Please check if you have errors in your logs.
You can also check the rabbitMQ stats directly in rabbit.

**Comment by xme** [comment]
Indeed I've something:

`{"user":{"ip":"::ffff:172.22.0.2","user_id":"88ec0c6a-13ce-5e39-b486-354fe4a7084f","referer":"https://xxxxx/dashboard/data/connectors"},"type":"READ_ERROR","operation_query":"query WorkersStatusQuery{...WorkersStatus_data}fragment WorkersStatus_data on Query{rabbitMQMetrics{consumers overview{queue_totals{messages messages_ready messages_unacknowledged}message_stats{ack ack_details{rate}}}}}","inner_relation_creation":0,"operation":"WorkersStatusQuery","time":38530,"variables":{},"size":2,"error":{"stacktrace":["Error: socket hang up","at connResetException (node:internal/errors:642:14)","at Socket.socketOnEnd (node:_http_client:486:23)","at Socket.emit (node:events:390:22)","at endReadableNT (node:internal/streams/readable:1307:12)","at processTicksAndRejections (node:internal/process/task_queues:81:21)"]},"category":"APP","version":"4.5.4","level":"error","message":"API Call","timestamp":"2021-06-09T07:24:13.864Z"}`

Also, this seems strange, the number of messages in the queues are empty when listing queues from the rabbitmq container:

```
root@912e4e8645a1:/# rabbitmqadmin --user=rmquser --password=xxxxx list queues
+---------------------------------------------+----------+
|                    name                     | messages |
+---------------------------------------------+----------+
| listen_xxxxxxxx-430e-4bd0-8239-a83b7466d6f9 |          |
| listen_xxxxxxxx-e5fb-4508-aa17-17d042f3fc82 |          |
| listen_xxxxxxxx-86c9-4fc6-a4cf-a2dad8abdb28 |          |
| listen_xxxxxxxx-95c1-4c12-b4b0-77583f50e7af |          |
| listen_xxxxxxxx-a5db-41e4-8986-4f6e7ea7a7ed |          |
| listen_xxxxxxxx-2a9f-4ae2-b69b-b0933b9b770a |          |
| listen_xxxxxxxx-d020-4904-8217-9597d0789603 |          |
| listen_xxxxxxxx-117f-49ec-921c-ca909adf0427 |          |
| listen_xxxxxxxx-ffcd-437f-b239-73cd801a0baa |          |
| listen_xxxxxxxx-d3ab-45dd-b333-b5fc36a87014 |          |
| listen_xxxxxxxx-37ab-4b99-afd2-eb8c34129b33 |          |
| listen_xxxxxxxx-f099-4172-991b-f2730f4b531d |          |
| listen_xxxxxxxx-bc00-4f41-9f24-b7c8c18d9237 |          |
| listen_xxxxxxxx-4eaf-4fe2-9fd1-658c8a4def62 |          |
| listen_xxxxxxxx-751b-45c8-a76a-4d12026c9cf4 |          |
| push_xxxxxxxx-430e-4bd0-8239-a83b7466d6f9   |          |
| push_xxxxxxxx-e5fb-4508-aa17-17d042f3fc82   |          |
| push_xxxxxxxx-86c9-4fc6-a4cf-a2dad8abdb28   |          |
| push_xxxxxxxx-95c1-4c12-b4b0-77583f50e7af   |          |
| push_xxxxxxxx-a5db-41e4-8986-4f6e7ea7a7ed   |          |
| push_xxxxxxxx-2a9f-4ae2-b69b-b0933b9b770a   |          |
| push_xxxxxxxx-d020-4904-8217-9597d0789603   |          |
| push_xxxxxxxx-117f-49ec-921c-ca909adf0427   |          |
| push_xxxxxxxx-ffcd-437f-b239-73cd801a0baa   |          |
| push_xxxxxxxx-d3ab-45dd-b333-b5fc36a87014   |          |
| push_xxxxxxxx-37ab-4b99-afd2-eb8c34129b33   |          |
| push_xxxxxxxx-f099-4172-991b-f2730f4b531d   |          |
| push_xxxxxxxx-bc00-4f41-9f24-b7c8c18d9237   |          |
| push_xxxxxxxx-4eaf-4fe2-9fd1-658c8a4def62   |          |
| push_xxxxxxxx-751b-45c8-a76a-4d12026c9cf4   |          |
+---------------------------------------------+----------+
```
But ingestion is working!?

**Comment by richard-julien** [comment]
Maybe rabbitMQ is not a good condition :) Can you try to restart it?

**Comment by xme** [comment]
Already done, but I'll try again... RabiitMQ does not report any error in logs

**Comment by xme** [comment]
So, I restarted the rabbitmq, worker, same problem.
When I restarted the worker after the upgrade, there was some missing queues:

`WARNING:pika.channel:Received remote Channel.Close (404): "NOT_FOUND - no queue 'push_xxxxxxxx-2a48-4fa2-b4b4-f13da42714fe' in vhost '/'" on <Channel number=1 OPEN conn=<SelectConnection OPEN transport=<pika.adapters.utils.io_services_utils._AsyncPlaintextTransport object at 0x7fb7f8c80370> params=<ConnectionParameters host=rabbitmq port=5672 virtual_host=/ ssl=False>>>`

I created them manually in rabbitmq via:

`# rabbitmqadmin -u rmquser -p xxx declare queue name=<queue_name>`

And again, restarted the containers... Still the same!

**Comment by xme** [comment]
Ok, found something interesting...
If I use "rabbitmqctl" to list queues -> I've found of messages
If I use "rabbitmqadm" -> No messages :)
But it does not explain why stats are not present in OpenCTI...

**Comment by richard-julien** [comment]
OpenCTI use the JSON API of the rabbitmq management plugin.
Do you still have "Error: socket hang up" errors in your logs?

**Comment by xme** [comment]
No more errors since the last one... The server is running smoothly and ingesting data...

**Comment by richard-julien** [comment]
Can you try to execute this query in the graphql playground (/graphql)

```
query stats {
  rabbitMQMetrics {
    queues {
      name
      messages
      messages_ready
      messages_unacknowledged
    }
    consumers
  }
}
```

**Comment by xme** [comment]
From a TCP/IP or API Point of view it works but I've 'null' values for all queues:

```
{
          "name": "push_xxxxxxxx-4eaf-4fe2-9fd1-658c8a4def62",
          "messages": null,
          "messages_ready": null,
          "messages_unacknowledged": null
},
```
Edited:
Consumers is also 0

**Comment by richard-julien** [comment]
Can you tell me the exact version of the rabbitmq and management plugin you install/deploy?

**Comment by xme** [comment]
rabbitmq:3.8-management
opencti/platform:4.5.4
opencti/worker:4.5.4
rabbitmqadmin 3.8.11

**Comment by xme** [comment]
Any news? Where you able to reproduce the problem?

**Comment by richard-julien** [comment]
Hi @xme . No sorry still not able to reproduce on my side.

**Comment by ghost** [comment]
Hi, I have the same error.  
has anyone found a solution? thx :)

![image](https://user-images.githubusercontent.com/26391831/122761039-13920580-d29c-11eb-98c1-9f69e822747a.png)

```json
{
   "user":{
      "ip":"::ffff:172.18.0.2",
      "user_id":"88ec0c6a-13ce-5e39-b486-354fe4a7084f",
      "referer":"http://opencti/opencti/dashboard/data/connectors"
   },
   "type":"READ_ERROR",
   "operation_query":"query WorkersStatusQuery{...WorkersStatus_data}fragment WorkersStatus_data on Query{elasticSearchMetrics{docs{count}search{query_total fetch_total}indexing{index_total delete_total}get{total}}rabbitMQMetrics{consumers overview{queue_totals{messages messages_ready messages_unacknowledged}message_stats{ack ack_details{rate}}}}}",
   "inner_relation_creation":0,
   "operation":"WorkersStatusQuery",
   "time":71,
   "variables":{
      
   },
   "size":2,
   "error":{
      "stacktrace":[
         "GraphQLError: Int cannot represent non 32-bit signed integer value: 3009072338",
         "at GraphQLScalarType.serializeInt [as serialize] (/opt/opencti/node_modules/graphql/type/scalars.js:53:11)",
         "at completeLeafValue (/opt/opencti/node_modules/graphql/execution/execute.js:653:37)",
         "at completeValue (/opt/opencti/node_modules/graphql/execution/execute.js:578:12)",
         "at resolveField (/opt/opencti/node_modules/graphql/execution/execute.js:472:19)",
         "at executeFields (/opt/opencti/node_modules/graphql/execution/execute.js:292:18)",
         "at collectAndExecuteSubfields (/opt/opencti/node_modules/graphql/execution/execute.js:748:10)",
         "at completeObjectValue (/opt/opencti/node_modules/graphql/execution/execute.js:738:10)",
         "at completeValue (/opt/opencti/node_modules/graphql/execution/execute.js:590:12)",
         "at resolveField (/opt/opencti/node_modules/graphql/execution/execute.js:472:19)",
         "at executeFields (/opt/opencti/node_modules/graphql/execution/execute.js:292:18)",
         "at collectAndExecuteSubfields (/opt/opencti/node_modules/graphql/execution/execute.js:748:10)",
         "at completeObjectValue (/opt/opencti/node_modules/graphql/execution/execute.js:738:10)",
         "at completeValue (/opt/opencti/node_modules/graphql/execution/execute.js:590:12)",
         "at /opt/opencti/node_modules/graphql/execution/execute.js:469:16",
         "at runMicrotasks (<anonymous>)",
         "at processTicksAndRejections (node:internal/process/task_queues:94:5)"
      ]
   },
   "category":"APP",
   "version":"4.5.5",
   "level":"error",
   "message":"API Call",
   "timestamp":"2021-06-21T08:56:12.239Z"
}

```

**Comment by SamuelHassine** [comment]
This has been solved in 5.0.0.
